### DISCUSSION TITLE
QTree: let expandAll() and setExpanded(key, state) return a promise

### DISCUSSION THREAD

**Discussion by drahkrub** [quasarframework/quasar#13100] [Ideas / Proposals]
...(instead of `void 0`) which resolves when all animations are finished.

Use case: I've placed a (potentially) large QTree inside a QScrollArea as already described in https://github.com/quasarframework/quasar/discussions/13094. If some new node is added to the tree I would like to scroll this node into view. Unfortunately if the new node is a leaf node of a deeper nested structure which is added to the tree model than the new (parent) nodes are initially collapsed. Therefore I expand all parents of the newly attached leaf node up to the root. Only when this is completed (including all animations) the leaf node can be scrolled into view (otherwise I fail to calculate the scroll offset).

I guess there are also simpler use cases ;-) but they all boil down to "I want to wait for a stable version of the QTree = all animations are finished before ...".

Would be nice if this would be integrated into the already great Quasar! :-)

**Comment by MaximilianKresse** [comment]
To add my point of view on this why I need this feature: I have a tree with lazy loading and want to reload the tree. All open nodes should stay open or at least reopen to the previous state. A promise return on setExpanded would help greatly to solve this in a more elegant way.
